covid19_sfbayarea/data/utils.py

```diff
@@ -25,9 +26,18 @@ def __init__(self, base_url: str):
         self.metadata_url = urljoin(self.base_url, '/api/views/metadata/v1/')
 
     def request(self, url:str, **kwargs: Any) -> Dict:
-        response = self.session.get(url, **kwargs)
-        response.raise_for_status()
-        return response.json()
+        try:
+            response = self.session.get(url, **kwargs)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as http_err:
+            if response.raise_for_status == 400:
+                raise BadRequest(f'{http_err} \n Result from server: {response.json()}')
+                return response.json()
+            else:
+                print(f'{http_err}') # log the error
+                return dict() # return empty dict
+        else:
+            return response.json()
```

Comment:
So, a few thoughts here:

- `response.raise_for_status` should always be a bound function, not a status code (for that, use `response.status_code`), so the `if` statement here will never work right.

- The `return` immediately after the `raise`, never actually run.

- This probably shouldn’t be logging anything.

What if something more like:

```suggestion
        try:
            response = self.session.get(url, **kwargs)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.HTTPError as http_err:
            try:
                server_message = response.json()['message']
            except Exception:
                # There's no JSON data here, so re-raise the original error
                raise http_err
            raise BadRequest(f'{server_message} (status: {response.status_code}')
```

As a bonus, you could define `BadRequest` as a subclass of `requests.exceptions.HTTPError` and make it a bit more fancy:

```py
class BadRequest(requests.exceptions.HTTPError):
    """
    Represents a detailed error message from a web server.
    """
    def __str__(self):
        message = super().__str__()
        return f'{message} (status: {self.response.status_code})'

# Then instead of using like:
raise BadRequest(f'{server_message} (status: {response.status_code}')
# You can do:
raise BadRequest(server_message, response=response)
# And other code can also handle it like a normal exception from requests
# because it's a subclass
```